### PR TITLE
tests: Refactor agnhost image pod usage - common (part 1)

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

A previous commit added  a few agnhost related functions that creates agnhost pods / containers for general purposes.

Refactors tests to use those functions.

**Which issue(s) this PR fixes**:

Fixes #
Depends On: #87266

**Special notes for your reviewer**:

/cc @alejandrox1 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by alejandrox1 — COMMENTED** [review]
Left a comment but overall looking good.

File: test/e2e/common/kubelet_etc_hosts.go [hunk 1]
```diff
@@ -222,6 +176,8 @@ func (config *KubeletManagedHostConfig) createPodSpec(podName string) *v1.Pod {
 			},
 		},
 	}
+
+	pod.Spec.Containers[2].VolumeMounts = append([]v1.VolumeMount{{Name: "host-etc-hosts", MountPath: etcHostsPath}}, pod.Spec.Containers[2].VolumeMounts[0])
```
[the remark this answers is not in the document]
Comment (claudiubelu) [reply test/e2e/common/kubelet_etc_hosts.go]: Done

**Review by alejandrox1 — COMMENTED** [review]
/lgtm
/approve

Thank you @claudiubelu for working on this!

**Pull request merged**